File: server/models/account.rb
DB.create_table? :accounts do
  uuid :uuid, primary_key: true
  foreign_key :user_uuid, :users, null: false, type: 'uuid'
  String :name, null: false, unique: true
  DateTime :created_at, null: false
  DateTime :updated_at, null: false
end

class Account < Sequel::Model
  many_to_one :user, key: :user_uuid
end

File: packages/inab-server/src/database/entities/Account.js
import {
  Column,
  CreateDateColumn,
  Entity,
  ManyToOne,
  OneToMany,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
} from 'typeorm'
import {Transaction} from './Transaction'
import {User} from './User'

@Entity()
export class Account {
  @PrimaryGeneratedColumn('uuid') uuid = undefined

  @ManyToOne(() => User, user => user.accounts, {nullable: false})
  user = undefined

  @OneToMany(() => Transaction, transaction => transaction.account)
  transactions = undefined

  @Column('varchar') name = undefined

  @CreateDateColumn() created_at = undefined

  @UpdateDateColumn() updated_at = undefined
}
